### DISCUSSION TITLE
Move Adjustments before Tonemapping

### DISCUSSION THREAD

**Discussion by catboy-catfish** [godotengine/godot-proposals#9427] [3D]
I want to use the ACES tonemapper because of its realistic way of handling bright colors, but it adds too much contrast. Adjustments don't help, decreasing the contrast makes the image look washed out. This is because Adjustments comes after Tonemapping.

**Comment by Calinou** [comment]
- Related to https://github.com/godotengine/godot-proposals/issues/7263.

We can't change the default behavior to avoid breaking visuals in existing projects, but we could expose an Environment property to define the order in which operations are performed (from first to last):

- **Tonemap > Adjustments > Color Correction** (default, and the current behavior)
- **Tonemap > Color Correction > Adjustments**
- **Adjustments > Tonemap > Color Correction**
- **Adjustments > Color Correction > Tonemap**
- **Color Correction > Tonemap > Adjustments**
- **Color Correction > Adjustments > Tonemap**

> I want to use the ACES tonemapper because of its realistic way of handling bright colors, but it adds too much contrast. 

You may be interested in [AgX](https://github.com/godotengine/godot-proposals/discussions/7545) to avoid ACES' excessive contrast issues :slightly_smiling_face:

**Reply by catboy-catfish** [reply]
The exposed Environment property looks good. I can get behind that.

AgX sounds like a decent alternative, but obviously that's not an easy thing to implement. Not sure if LUTs can perform realistic desaturation of brighter objects like this, but I don't know much about LUTs anyway.

**Reply by Calinou** [reply]
> AgX sounds like a decent alternative, but obviously that's not an easy thing to implement.

I have a PR for it: https://github.com/godotengine/godot/pull/87260

> Not sure if LUTs can perform realistic desaturation of brighter objects like this, but I don't know much about LUTs anyway.

LUTs can do anything, including acting as a tonemapper if they are processed before tonemapping and you use Linear tonemapping in Environment.
